src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,374 @@
+    } catch (error) {
+      setFetchSwapQuoteError(true)
+      Logger.error(TAG, `Error fetching token swap quote: ${error}`)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  // Get swap info on load
+  useAsync(getSwapInfo, [])
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader title={t('swapReviewScreen.title')} left={<BackButton />} />
+      <DisconnectBanner />
+      {(loading && swapInfo === null) || fetchSwapQuoteError ? (
+        <View style={styles.loadingContentContainer}>
+          <ActivityIndicator
+            size="large"
+            color={colors.greenBrand}
+            testID="SwapReviewScreen/loading"
+          />
+        </View>
+      ) : (
+        <ScrollView
+          style={styles.contentContainer}
+          refreshControl={
+            <RefreshControl
+              tintColor="transparent"
+              colors={['transparent']}
+              style={{ backgroundColor: 'transparent' }}
+              refreshing={loading}
+              onRefresh={getSwapInfo}
+            />
+          }
+        >
+          {loading && (
+            <View style={styles.loadingContentContainer}>
+              <ActivityIndicator
+                size="large"
+                color={colors.greenBrand}
+                testID="SwapReviewScreen/loading"
+              />
+            </View>
+          )}
+          <View style={styles.subContentContainer}>
+            <View style={styles.tallRow}>
+              <Text style={styles.label}>{t('swapReviewScreen.swapFrom')}</Text>
+              <View style={styles.tokenDisplayView}>
+                <TokenDisplay
+                  style={styles.amountText}
+                  amount={divideByWei(swapInfo?.unvalidatedSwapTransaction?.sellAmount)}
+                  tokenAddress={fromToken}
+                  showLocalAmount={false}
+                  testID={'FromSwapAmountToken'}
+                />
+                <TokenDisplay
+                  style={styles.amountSubText}
+                  amount={divideByWei(swapInfo?.unvalidatedSwapTransaction?.sellAmount)}
+                  tokenAddress={fromToken}
+                  showLocalAmount={true}
+                  testID={`FromSwapAmountTokenLocal`}
+                />
+              </View>
+            </View>
+            <View style={styles.tallRow}>
+              <Text style={styles.label}>{t('swapReviewScreen.swapTo')}</Text>
+              <View style={styles.tokenDisplayView}>
+                <TokenDisplay
+                  style={[styles.amountText, { color: colors.greenUI }]}
+                  amount={divideByWei(
+                    swapInfo?.unvalidatedSwapTransaction?.buyAmount -
+                      swapInfo?.unvalidatedSwapTransaction?.gas
+                  )}
+                  tokenAddress={toToken}
+                  showLocalAmount={false}
+                  testID={'ToSwapAmountToken'}
+                />
+                <TouchableOpacity
```

Comment:
can we use `Touchable` here? you may need to add a fragment around the children

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2767/commits/d84c6544c21453e841b1e21842f0d25e2d211f22!